.github/workflows/auto-label.yml

```diff
@@ -0,0 +1,30 @@
+name: Auto-Label for Write Access and Python Files
+on:
+  pull_request:
+    types: [opened]
+jobs:
+  auto-label:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout the PR code
+      uses: actions/checkout@v3
+    - name: Check if the PR author has write access
+      id: write-access
+      uses: octokit/request-action@v3.0.0
+      with:
+        route: GET /repos/${{ github.repository }}/collaborators/${{ github.event.pull_request.user.login }}/permission
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      continue-on-error: true
+    - name: Verify Python files are modified
+      id: python-files
+      run: |
+        git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} | grep '.py' || echo 'no-python'
+      continue-on-error: true
+    - name: Add label if conditions are met
+      if: ${{ steps.write-access.outputs.permission == 'write' && steps.python-files.outputs.result != 'no-python' }}
+      uses: actions-ecosystem/action-add-labels@v1
```

Comment:
:(
```
actions-ecosystem/action-add-labels@v1 is not allowed to be used in NVIDIA/NeMo-Curator. Actions in this workflow must be: within a repository that belongs to your Enterprise account, created by GitHub, verified in the GitHub Marketplace, or matching the following: NVIDIA/gitlab-answer-app/*, aws-actions/configure-aws-credentials@*, cinotify/github-action@*, codecov/codecov-action@*, conda-forge/automerge-action@*, conda-incubator/setup-miniconda@*, convictional/trigger-workflow-and-wait@*, crazy-max/ghaction-docker-meta@*, dangoslen/changelog-enforcer@*, EndBug/add-and-commit@*, jacobtomlinson/gha-anaconda-package-version@*, jacobtomlinson/gha-find-replace@*, peter-evans/create-pull-request@*, peter-evans/dockerhub-description@*, pre-commit/action@*, pypa/gh-action-pypi-publish@*, release-drafter/release-drafter@*, ruby/setup-ruby@*, softprops/action-gh-release@*, test-summary/action@*, the-coding-turtle/ga-yaml-parser@*, devcontainers/action@*, tj-actions/changed-files@*, nrwl/las...
```
I might be able to use something else (like `peter-evans/create-pull-request@*`?) or use a GitHub API approach like:
```
- name: Add label if conditions are met
  if: ${{ steps.write-access.outputs.permission == 'write' && steps.python-files.outputs.result != 'no-python' }}
  run: |
    curl -X POST \
      -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
      -H "Accept: application/vnd.github.v3+json" \
      https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.pull_request.number }}/labels \
      -d '{"labels":["Run GPU CI"]}'
```

Reply:
This feature is mostly for convenience. I can add it in a future PR if desired.

Reply:
Opened https://github.com/NVIDIA/NeMo-Curator/issues/259 for now.